### PR TITLE
New application: Gpredict

### PR DESCRIPTION
**Pull request by sielicki**
This is the first application I'm trying to package for flatpak and flathub distribution. Looking for input when it comes to the sandboxed permissions and cleanup stages. For now, I think it is acceptable.

Being tracked over here by the application dev, https://github.com/csete/gpredict/issues/119

Signed-off-by: Nicholas Sielicki <sielicki@yandex.com>

### CODE REVIEW

**Comment by AsciiWolf** [comment]
bot, build net.oz9aec.Gpredict

File: net.oz9aec.Gpredict.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+  "app-id": "net.oz9aec.Gpredict",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.28",
+  "sdk": "org.gnome.Sdk",
+  "command": "gpredict",
+  "finish-args": [
+    "--share=ipc", "--socket=x11",
+    "--socket=wayland",
+    "--share=network", "--filesystem=home"
```
Comment (AsciiWolf) [inline net.oz9aec.Gpredict.json]: Does it really need full home access? If it is for the _~./gpredict_ directory, you can use `--persist=.gpredict` for that.

Comment (sielicki) [reply net.oz9aec.Gpredict.json]: Some users might like to use custom data files from their home directory, but this is generally atypical usage. Application data is stored under `.config/Gpredict`, so I'll restrict it to there.

**Comment by AsciiWolf** [comment]
bot, build net.oz9aec.Gpredict

**Comment by AsciiWolf** [comment]
It [failed](https://flathub.org/builds/#/builders/2/builds/2183) to build because of some AppData-related issue.

**Comment by AsciiWolf** [comment]
bot, build net.oz9aec.Gpredict

**Comment by AsciiWolf** [comment]
There are still some issues with the AppData file:

```
Processing application net.oz9aec.Gpredict
gpredict.desktop:                  AppData problem: tag-missing : <metadata_license> is not present
gpredict.desktop:                  AppData problem: tag-invalid : <icon> not allowed in desktop appdata
```

**Comment by AsciiWolf** [comment]
`zgrep "<id>net.oz9aec.Gpredict\(.desktop\)\?</id>" builddir/*/share/app-info/xmls/net.oz9aec.Gpredict.xml.gz`
` in dir /srv/buildbot/worker/build-x86_64-7/net.oz9aec.Gpredict (timeout 1200 secs)
 watching logfiles {}`
` argv: zgrep "<id>net.oz9aec.Gpredict\(.desktop\)\?</id>" builddir/*/share/app-info/xmls/net.oz9aec.Gpredict.xml.gz`
` using PTY: False`
`program finished with exit code 1`

I think that you are missing `rename-desktop-file` in the json file.

File: net.oz9aec.Gpredict.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{
+	"app-id": "net.oz9aec.Gpredict",
+	"runtime": "org.gnome.Platform",
+	"runtime-version": "3.28",
+	"sdk": "org.gnome.Sdk",
+	"command": "gpredict",
+	"finish-args": [
+		"--share=ipc", "--socket=x11",
+		"--socket=wayland",
+		"--share=network", "--persist=.config/Gpredict"
```
Comment (barthalion) [inline net.oz9aec.Gpredict.json]: You don't need `--persist` here, appropriate XDG variable will be exported and app will be write there the config automatically.

File: net.oz9aec.Gpredict.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+			"sources": [
+				{
+					"type": "archive",
+					"url": "https://sourceforge.net/projects/hamlib/files/hamlib/3.2/hamlib-3.2.tar.gz",
+					"sha256": "b55cb5e6a8e876cceb86590c594ea5a6eb5dff2e30fc13ce053b46baa6d7ad1d"
+				}
+			]
+		},
+		{
+			"name": "appdata",
```
Comment (barthalion) [inline net.oz9aec.Gpredict.json]: Should be contributed upstream (and before that, it can be added just as `file` resource under gpredict).

**Comment by barthalion** [comment]
Have you contacted upstream author about adding their app to Flathub?

**Comment by nedrichards** [comment]
bot, build net.oz9aec.Gpredict

File: net.oz9aec.Gpredict.appdata.xml [hunk 4]
```diff
@@ -24,6 +24,10 @@
 <image type="source">https://comzeradd.fedorapeople.org/screenshots/gpredict1.png</image>
 </screenshot>
 </screenshots>
+	<releases>
+		<release version="2.2.1" date="2018-01-21"/>
+		<release version="2.2" date="2018-01-05"/>
+</releases>
```
Comment (AsciiWolf) [inline net.oz9aec.Gpredict.appdata.xml]: Small typo - missing indentation.

**Comment by AsciiWolf** [comment]
The [build](https://flathub.org/builds/#/builders/2/builds/2308) passed on all archs and everything (except the small typo in AppData file) seems to be ok.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/net.oz9aec.Gpredict

**Pull request closed without merging**